### PR TITLE
Set branch name on build info.

### PR DESCRIPTION
**Pull request by MitchellBurton**
This is a copy of the PR into the repo this was forked off of: https://github.com/xo-energy/action-octopus-build-information/pull/19

Well will leave this to sit for a few days, if the upstream PR is accepted then we can close this. If the upstream is not accepted then we will need to merge and tag this to make it stable for LIC use.

--------------------------------------

Adds the branch name to the Build info to be pushed to Octopus Deploy. 
If the ref starts with `refs/heads/` then this will strip that and set the branch name in the build info. If the ref doesn't start with `refs/heads/` then this will leave the branch name empty. This should cope with the case where builds are triggered on tags rather then branches.

Solves https://github.com/xo-energy/action-octopus-build-information/issues/17

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -286,13 +286,19 @@ async function run() {
     const commits = await getCommits(github, previousRef);
     core.info(`Collected ${commits.length} commits`);
 
+    let branch = "";
+    if (branch.startsWith("refs/heads/")) {
+      branch = context.ref.replace("refs/heads/", "");
```
Comment (cadon0) [inline src/main.js:291]: should it be this?
```suggestion
    if (context.ref.startsWith("refs/heads/")) {
      branch = context.ref.replace("refs/heads/", "");
```

**Review by cadon0 — COMMENTED** [review]
Their readme has an output sample which could be updated
image

File: src/main.js (hunk 1 above)
Comment (MitchellBurton) [reply src/main.js]: Yes, yes it should.

**Review by ewen — COMMENTED** [review]
Looks good

File: src/main.js [hunk 2]
```diff
@@ -286,13 +286,19 @@ async function run() {
     const commits = await getCommits(github, previousRef);
     core.info(`Collected ${commits.length} commits`);
 
+    let branch = "";
```
Comment (cadon0) [inline src/main.js:289]: might be nicer for consumers of this json if the default is `null`?

**Review by cadon0 — APPROVED** [review]

File: src/main.js (hunk 2 above)
Comment (MitchellBurton) [reply src/main.js]: Done.

**Pull request merged**